hasktorch/test/NNSpec.hs

```diff
@@ -16,6 +17,67 @@ import Torch.Autograd
 import Torch.NN
 import GHC.Generics
 
+data CustomlinearSpec = CustomlinearSpec
+  { in_features :: Int,
+    out_features :: Int
+  }
+  deriving (Show, Eq)
+
+data Customlinear = Customlinear
+  { customWeight :: Parameter,
+    customBias :: Parameter
+  }
+  deriving (Show, Generic)
+
+instance GTraversable Parameter Customlinear where
+  gflatten Customlinear {..} = [customWeight, customBias, customBias] -- These duplicated biases are for testing!!
+  gupdate _ = do
+    customWeight <- gpop
+    customBias <- gpop
+    _ <- gpop
+    return $ Customlinear {..}
```

Comment:
> 1. No Generic instance exists for a given f?
> 2. No Generic instance can be automatically derived for a given f?

We can implement a custom GTraversable instance with or without a generic instance.

> 3. A user wishes to implement a custom GTraversable instance for a given f?

I think so.
I think a user wishes to use a generic GTraversable instance for the time being.

Reply:
An additional general instance is not allowed. 
The following code causes  a duplicated instance error.
```
instance {-# OVERLAPS #-} (Scalar f) => GTraversable a f
```